### DISCUSSION TITLE
liveProvider not working on some pages

### DISCUSSION THREAD

**Discussion by vadim-roshuor** [refinedev/refine#1881] [General]
Hello!
I have page with products. And I want to connect live provider on this page. But liveProvider not working on this page, but working on Users page. 
Could it depend on requests to the server?
There are standard queries on the User page, and custom queries on the Product page

App:
 ```
            dataProvider={changedDataProvider()}
            liveProvider={ablyLiveProvider}
            liveMode='auto'
```
Products:
```
 const { tableQueryResult, current, pageSize, setCurrent, setPageSize } =
        useTable<WCProductsI>({
            metaData: {
                fields: ['id', 'name', 'sku', 'slug', 'price', 'dateModified'],
            }
        })
```

Hasura provider:
```
getList: async ({ resource, sort, filters, pagination, metaData }) => {
            const current = pagination?.current ?? 1
            const limit = pagination?.pageSize || 10
            const offset = (current - 1) * limit

            const hasuraSorting = generateSorting(sort)
            const hasuraFilters = generateFilters(filters)

            const operation = metaData?.operation ?? resource

            const aggregateOperation = `${operation}_aggregate`

            const hasuraSortingType = `[${operation}_order_by!]`
            const hasuraFiltersType = `${operation}_bool_exp`

            const variablesObj = operationObj[operation]
                ? {
                      page: current,
                      perPage: limit,
                  }
                : {
                      limit,
                      offset,
                      ...(hasuraSorting && {
                          order_by: {
                              value: hasuraSorting,
                              type: hasuraSortingType,
                          },
                      }),
                  }
            const aggregateOperationObj = operationObj[operation]
                ? ''
                : {
                      operation: aggregateOperation,
                      fields: [{ aggregate: ['count'] }],
                      variables: {
                          where: {
                              value: hasuraFilters,
                              type: hasuraFiltersType,
                          },
                      },
                  }

            const { query, variables } = gql.query([
                {
                    operation,
                    fields: metaData?.fields,
                    variables: variablesObj,
                },
                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                // @ts-ignore
                aggregateOperationObj,
            ])

            const result = await client.request(query, variables)

            const total = operationObj[operation]
                ? ''
                : result[aggregateOperation].aggregate.count

            return {
                data: result[operation],
                total,
            }
        },
```

Thank you

**Comment by omeraplak** [comment]
Hey @vadim-roshuor 👋,
Thank you for contacting us! I'm going to ask you to make a few changes so that we can find the error more easily.

Could you do the same for your "users" page, if possible? so we can compare. 

I will ask you to add `onLiveEvent` for `useTable` and `Refine` component.

pages;
```
  const { tableQueryResult, current, pageSize, setCurrent, setPageSize } =
         useTable<WCProductsI>({
             metaData: {
                 fields: ['id', 'name', 'sku', 'slug', 'price', 'dateModified'],
             }
             onLiveEvent: (event) => { console.log("event", { event }); }
         });
```

Refine component;
```
<Refine
....
...
            onLiveEvent={(event) => {
               console.log("event", { event });
            }}
        />
```
Then can you share the result with me? Thank you for your patience! Don't forget to come to our [Discord server](https://discord.gg/refine) for faster support!

**Reply by vadim-roshuor** [reply]
Thank you @omeraplak 
I added onLiveEvent on "users" and on "Refine" and got :
![Screenshot from 2022-06-02 15-25-08](https://user-images.githubusercontent.com/102044742/171628880-91d63ad5-fc45-4fcb-a286-fc35289d6723.png)

When I added same onLiveEvent on "products" I got nothing. my console was empty

**Reply by omeraplak** [reply]
I think the only way to find this error is to create the live provider by following the document and then add `console.log` to the `subscribe` and `publish` functions to try to understand the issue

https://refine.dev/docs/core/providers/live-provider/#creating-a-live-provider

source code: https://github.com/pankod/refine/blob/master/packages/ably/src/index.ts

**Reply by vadim-roshuor** [reply]
I did it 
users:
![Screenshot from 2022-06-02 16-52-20](https://user-images.githubusercontent.com/102044742/171645341-d09529a6-3c52-4771-bb9e-83cce850bccc.png)


products:
![Screenshot from 2022-06-02 16-53-53](https://user-images.githubusercontent.com/102044742/171645551-3e694e79-4dd4-43c4-96e0-8dfdb52ae329.png)

**Reply by omeraplak** [reply]
Users page looks good but I couldn't see a "publish" log for "products".  Did you do a mutation(create/edit/delete) for "products" and still no logs?

**Reply by vadim-roshuor** [reply]
if I do a mutation I get this:
![Screenshot from 2022-06-02 17-18-21](https://user-images.githubusercontent.com/102044742/171650611-d5587b26-904b-45f0-b6f1-a496f6a2d7a6.png)

only publish logs

**Reply by omeraplak** [reply]
Finally, we detected the problem. When subscribing to live events, you are using the name `productsList`, but the resource name of your mutations is `updateProduct`. Refine cannot manage state automatically when these two are not the same

What you can do in this situation,
 
Using the **same** resource names on `mutations` and `queries`. If this is not possible, you have to manage this process yourself. You can use the [useSubscription](https://refine.dev/docs/core/hooks/live/useSubscription/) hook for this,


example,
```
  const { tableQueryResult, current, pageSize, setCurrent, setPageSize } =
         useTable<WCProductsI>({
             metaData: {
                 fields: ['id', 'name', 'sku', 'slug', 'price', 'dateModified'],
             }
         });

    useSubscription({
        channel: "resources/updateProduct",
        types: ["*"],
        onLiveEvent: (event) => {
            // refetch table data
            tableQueryResult.refetch();
        },
    });
```

I hope it will be useful for you

**Reply by vadim-roshuor** [reply]
thank you very much. you saved me

**Reply by omeraplak** [reply]
You're welcome. Don't forget to support us 🚀
